FirebaseDatabase/Sources/Utilities/FNextPushId.m

```diff
@@ -103,4 +103,40 @@ + (NSString *)successor:(NSString *_Nonnull)key {
     return [next substringWithRange:NSMakeRange(0, i + 1)];
 }
 
+// `key` is assumed to be non-empty.
++ (NSString *)predecessor:(NSString *_Nonnull)key {
```

Comment:
Can you add a unit test for just this method? I think it's complicated enough to warrant its own tests.